src/linker/Linker.Steps/ResolveFromAssemblyStep.cs

```diff
@@ -120,22 +123,29 @@ public static void ProcessLibrary (LinkContext context, AssemblyDefinition assem
 						// Both cases are bugs not on our end but we still want to link all assemblies
 						// especially when such types cannot be used anyway
 						//
-						context.LogMessage ($"Cannot find declaration of exported type '{exported}' from the assembly '{assembly}'");
+						context.LogMessage ($"Cannot find declaration of exported type '{exported}' from the assembly '{_assembly}'");
 
 						continue;
 					}
 
 					context.Resolve (resolvedExportedType.Scope);
 					MarkType (context, resolvedExportedType, rootVisibility);
-					context.MarkingHelpers.MarkExportedType (exported, assembly.MainModule);
+					context.MarkingHelpers.MarkExportedType (exported, _assembly.MainModule);
 				}
 			}
 
 			context.Tracer.Pop ();
 		}
 
-		static void MarkType (LinkContext context, TypeDefinition type, RootVisibility rootVisibility)
+		bool MarkType (LinkContext context, TypeDefinition type, RootVisibility rootVisibility)
 		{
+			if (type.IsInterface) {
+				if (_markedInterfaces == null)
```

Comment:
I don't like this logic hear, we are spreading Marking logic into a different step which will be hard to keep in sync

Reply:
I think this is easier to maintain than having MarkStep deal with both the public surface area "roots" and the conditional interface marking - see my other comment. LMK what you think.